### PR TITLE
(doc) Add release notes for CLI 2.4.0

### PR DESCRIPTION
**Pull request by vexx32**
## Description Of Changes

- Add release notes for CLI v2.4.0

## Motivation and Context

New release, new notes.

## Testing

* [x] I have previewed these changes using the [Docker Container](https://github.com/chocolatey/docs/tree/master/.devcontainer) or another method before submitting this pull request.

## Change Types Made

* [ ] Minor documentation fix (typos etc.).
* [x] Major documentation change (refactoring, reformatting or adding documentation to existing page).
* [ ] New documentation page added.
* [ ] The change I have made should have a video added, and I have raised an issue for this.
    * Issue #

## Change Checklist

* [ ] Requires a change to menu structure (top or left-hand side)/
* [x] Menu structure has been updated (highlights updated)

## Related Issue

N/A

### CODE REVIEW

File: src/content/docs/en-us/choco/release-notes.mdx [hunk 1]
```diff
@@ -24,6 +24,35 @@ This covers changes for the "chocolatey" and "chocolatey.lib" packages, which ar
...
+- Fix - Ensure .arguments files that cannot be decrypted do not prevent running CLI commands - see [#3503](https://github.com/chocolatey/choco/issues/3503).
+- Fix - Ensure invalid .arguments files do not prevent running CLI commands - see [#3513](https://github.com/chocolatey/choco/issues/3513).
+
+### Deprecated Feature
+
+- Deprecate `Install-ChocolateyPinnedTaskBarItem` - see [#3520](https://github.com/chocolatey/choco/issues/3520).
+
+### Documentation
+
+- (doc) Update build instructions - see [#3467](https://github.com/chocolatey/choco/pull/3467).
```
Comment (gep13) [inline src/content/docs/en-us/choco/release-notes.mdx:45]: The (doc) is not needed at the start here, as already under `Documentation` section.

File: src/content/docs/en-us/choco/release-notes.mdx [hunk 2]
```diff
@@ -24,6 +24,35 @@ This covers changes for the "chocolatey" and "chocolatey.lib" packages, which ar
...
+- (doc) Fixed parameter name typo in the chocolateyInstall template - see [#3504](https://github.com/chocolatey/choco/pull/3504).
```
Comment (gep13) [inline src/content/docs/en-us/choco/release-notes.mdx:46]: The (doc) is not needed at the start here, as already under `Documentation` section.

File: src/content/docs/en-us/choco/release-notes.mdx (hunk 1 above)
Comment (vexx32) [reply src/content/docs/en-us/choco/release-notes.mdx]: Fixed!

File: src/content/docs/en-us/choco/release-notes.mdx (hunk 2 above)
Comment (vexx32) [reply src/content/docs/en-us/choco/release-notes.mdx]: Fixed!

**Review by gep13 — CHANGES_REQUESTED** [review]

File: src/content/docs/en-us/choco/release-notes.mdx [hunk 3]
```diff
@@ -24,6 +24,35 @@ This covers changes for the "chocolatey" and "chocolatey.lib" packages, which ar
 
 <ChocolateyComponentDependenciesLink />
 
+## 2.4.0 (Nov 12, 2024) \{#v2.4.0}
+
+### Bug Fixes
+
+- Fix - Update tab completions for compatibility with PowerShell 7.4+ - see [#3364](https://github.com/chocolatey/choco/issues/3364).
+- Fix - Incompatible meta/install packages lead to unnecessary downgrade - see [#3461](https://github.com/chocolatey/choco/issues/3461).
```
[the remark this answers is not in the document]
Comment (vexx32) [reply src/content/docs/en-us/choco/release-notes.mdx:32]: I think the best succinct way to put it is... probably `Prevent dependency resolution from unexpectedly downgrading packages`?

[the remark this answers is not in the document]
Comment (gep13) [reply src/content/docs/en-us/choco/release-notes.mdx]: Yes, this sounds good to me!

File: src/content/docs/en-us/choco/release-notes.mdx [hunk 4]
```diff
@@ -24,6 +24,35 @@ This covers changes for the "chocolatey" and "chocolatey.lib" packages, which ar
...
+
+### Bug Fixes
+
+- Fix - Update tab completions for compatibility with PowerShell 7.4+ - see [#3364](https://github.com/chocolatey/choco/issues/3364).
+- Fix - Prevent dependency resolution from unexpectedly downgrading packages - see [#3461](https://github.com/chocolatey/choco/issues/3461).
+- Fix - Do not install packages if dependencies fail to install - see [#3487](https://github.com/chocolatey/choco/issues/3487).
+- Fix - Ensure package folders are correctly restored if unexpected failures occur during upgrade - see [#3489](https://github.com/chocolatey/choco/issues/3489).
+- Fix - Use the user-provided page size when querying repositories - see [#3502](https://github.com/chocolatey/choco/issues/3502).
+- Fix - Ensure `.arguments` files that cannot be decrypted do not prevent running CLI commands - see [#3503](https://github.com/chocolatey/choco/issues/3503).
+- Fix - Ensure invalid `.arguments` files do not prevent running CLI commands - see [#3513](https://github.com/chocolatey/choco/issues/3513).
```
Comment (gep13) [inline src/content/docs/en-us/choco/release-notes.mdx:37]: Apologies, this was meant to be in the last review.

```suggestion
- Fix - Ensure invalid `.arguments` files do not prevent running Chocolatey CLI commands - see [#3513](https://github.com/chocolatey/choco/issues/3513).
```

File: src/content/docs/en-us/choco/release-notes.mdx [hunk 5]
```diff
@@ -24,6 +24,35 @@ This covers changes for the "chocolatey" and "chocolatey.lib" packages, which ar
...
+## 2.4.0 (Nov 12, 2024) \{#v2.4.0}
+
+### Bug Fixes
+
+- Fix - Update tab completions for compatibility with PowerShell 7.4+ - see [#3364](https://github.com/chocolatey/choco/issues/3364).
+- Fix - Prevent dependency resolution from unexpectedly downgrading packages - see [#3461](https://github.com/chocolatey/choco/issues/3461).
+- Fix - Do not install packages if dependencies fail to install - see [#3487](https://github.com/chocolatey/choco/issues/3487).
+- Fix - Ensure package folders are correctly restored if unexpected failures occur during upgrade - see [#3489](https://github.com/chocolatey/choco/issues/3489).
+- Fix - Use the user-provided page size when querying repositories - see [#3502](https://github.com/chocolatey/choco/issues/3502).
+- Fix - Ensure `.arguments` files that cannot be decrypted do not prevent running CLI commands - see [#3503](https://github.com/chocolatey/choco/issues/3503).
```
Comment (gep13) [inline src/content/docs/en-us/choco/release-notes.mdx:36]: Apologies, this was meant to be in the last review.

```suggestion
- Fix - Ensure `.arguments` files that cannot be decrypted do not prevent running Chocolatey CLI commands - see [#3503](https://github.com/chocolatey/choco/issues/3503).
```

**Review by gep13 — CHANGES_REQUESTED** [review]
One minor thing in a couple of places.

Also, so correctness, can you also update the GitHub issue titles with the changes that have been made in the release notes?

Thanks!

**Review by gep13 — APPROVED** [review]
LGTM!

**Comment by gep13** [comment]
@vexx32 thank you for getting these pulled together!

**Pull request merged**